### ISSUE TITLE
Farming Mode: Only keep items on character that were there when farming mode started

### ISSUE THREAD

**Issue by danrspencer** [DestinyItemManager/DIM#8756]
### Proposed change

I'd like to be enable a version of farming mode which automatically moves away any item that was not in my inventory when the mode was turned on. Item's moved via DIM would be added / subtracted to the known inventory but items added via any other means would be considered a drop and moved to the vault.

### How does this fit into your workflow?

Often I have a few guns I want to carry around and switch between, but when I go to equip them in game they've been moved away to the vault. This mode would be an attempt to have a clean in game inventory of "hey, this is the stuff I actually care about and want to carry around" while automatically storing things that are picked up. 

It would also work well with loadouts that include guns to be added to inventory rather than equipped by preventing those weapons from being moved away.

**Comment by danrspencer** [comment]
I am willing to work on this myself if someone can give me a nudge in the right direction

**Comment by bhollis** [comment]
As a note, anything manually moved or tagged will be the last to be moved away. Consider putting the items you like to use in a loadout, so you can recall them any time.

**Comment by danrspencer** [comment]
Interesting, I didn’t know that.

I think this feature would still be useful though. “Weapons I want to carry” and “random stuff I picked up from the floor” are two completely separate concerns and I hate how intermingled they are in D2 inventory. 

It sounds the existing functionality would be good news for this feature too, something must already be tracking the difference between “weapons I put here” and “stuff I picked up”. 🤔
